### DISCUSSION TITLE
Error 409: Can't save the changes. Someone else is currently editing the same file

### DISCUSSION THREAD

**Discussion by beshur** [crowdin/crowdin-api-client-js#176] [Q&A] [answered]
Hi.

I build an API script that takes changes to multiple files and uploads them line by line using API methods.

However looks like when I have multiple lines one after another the second request fails with error: 
> Error: Can't save the changes. Someone else is currently editing the same file

What is behind this behaviour? How should I update multiple lines?

I also did not find details about this case in docs.

Thank you.

**Comment by andrii-bodnar** [comment, marked as answer]
@beshur hi, could you please clarify what API method you are using?

**Reply by beshur** [reply]
It happened with `addString`

**Reply by andrii-bodnar** [reply]
@beshur every string adding - it's a file update. Updates can't happen in parallel that's why there is such a 409 error. Please try to avoid parallel strings adding and do this less frequently - for example, every second. This time depends on file size. We are going to speed up this process in the nearest future.

If possible, the best way is to use the [updateOrRestoreFile](https://crowdin.github.io/crowdin-api-client-js/classes/SourceFiles.html#updateOrRestoreFile) method for massive updates instead of calling the addString for each string.

**Reply by beshur** [reply]
Thank you for reply.

Well, we wanted to use atomic `addString` method specifically to not replace the file completely in order to prevent the overwrites by different team members.

Looks like I could fix the issue by running the API calls sequentially and adding a 50ms delay between them.
